rivescript/parser.py

```diff
@@ -574,6 +574,14 @@ def check_syntax(self, cmd, line):
                     angle += 1
                 elif char == '>':
                     angle -= 1
+                elif char == '|':
+                    if parens == 0 and square == 0:   # Pipe outside the alternative and option
+                        return "Pipe | must be within parenthesis brackets or square brackets"
+
+                total = parens + square + curly + angle   # At each character, not more than 1 bracket opens
+                for special_char_count in [parens, square, curly, angle, total]:
+                    if special_char_count not in (0, 1):
+                        return "Unbalanced brackets"
```

Comment:
I actually wasn't aware of that usage of angle brackets. Does `{}` only apply to e.g. `{weight=100}`? Or I can assume that  curly brackets would not be nested in other brackets (`[] ()`). If so, I will simply remove angle from the total, which means that `[, ( {` can only open one at a time.